pkg/vault/vault_interface.go

```diff
@@ -0,0 +1,321 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package vault
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/api"
+)
+
+var httpClient = &http.Client{
+	Timeout: 10 * time.Second,
+}
+
+// Connection contains required information for connecting to vault
+type Connection struct {
+	Client  *api.Client
+	Address string
+	Token   string
+}
+
+// Dummy implementation for testing
+type Dummy struct {
+}
+
+// NewConnection returns a new Connection object
+func NewConnection(c *api.Client, addr string, token string) *Connection {
+	return &Connection{
+		Client: c,
+		Address: addr,
+		Token: token,
+	}
+}
+
+// NewDummyConnection returns a new Dummy object
+func NewDummyConnection() *Dummy {
+	return &Dummy{}
+}
+
+// CredentialManagerInterface provides vault functionality
+type CredentialManagerInterface interface {
```

Comment:
consider moving this interface to another file as its not vault specific. or rename the package name to something like secretmanager as it contains the dummy implementation as well.

Reply:
Split into several files, renamed the interface name